test/libsolidity/SemanticTest.cpp

```
@@ -175,122 +177,109 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 					soltestAssert(deploy("", 0, bytes(), libraries), "Failed to deploy contract.");
 				constructed = true;
 			}
-
-			if (test.call().kind == FunctionCall::Kind::Storage)
-			{
-				test.setFailure(false);
-				bytes result(1, !storageEmpty(m_contractAddress));
-				test.setRawBytes(result);
-				soltestAssert(test.call().expectations.rawBytes().size() == 1, "");
-				if (test.call().expectations.rawBytes() != result)
-					success = false;
-			}
-			else if (test.call().kind == FunctionCall::Kind::Constructor)
-			{
-				if (m_transactionSuccessful == test.call().expectations.failure)
-					success = false;
-
-				test.setFailure(!m_transactionSuccessful);
-				test.setRawBytes(bytes());
-			}
-			else
-			{
-				bytes output;
-				if (test.call().kind == FunctionCall::Kind::LowLevel)
-					output = callLowLevel(test.call().arguments.rawBytes(), test.call().value.value);
-				else
-				{
-					soltestAssert(
-						m_allowNonExistingFunctions ||
-						m_compiler.methodIdentifiers(m_compiler.lastContractName()).isMember(test.call().signature),
-						"The function " + test.call().signature + " is not known to the compiler"
-					);
-
-					output = callContractFunctionWithValueNoEncoding(
-						test.call().signature,
-						test.call().value.value,
-						test.call().arguments.rawBytes()
-					);
-				}
-
-				if (
-					m_transactionSuccessful == test.call().expectations.failure ||
-					output != test.call().expectations.rawBytes()
-				)
-					success = false;
-
-				test.setFailure(!m_transactionSuccessful);
-				test.setRawBytes(std::move(output));
-				test.setContractABI(m_compiler.contractABI(m_compiler.lastContractName()));
-			}
+		}
+		catch (langutil::UnimplementedFeatureError const&)
+		{
+			// ignore unimplemented feature errors when forcibly trying to compile via yul.
+			if (!_compileViaYul || m_runWithYul)
+				throw;
```

Also why not rethrow the unimplemented error?